util/clusterauth/clusterauth.go

```diff
@@ -215,39 +216,103 @@ func InstallClusterManagerRBAC(clientset kubernetes.Interface, ns string, namesp
 	return GetServiceAccountBearerToken(clientset, ns, ArgoCDManagerServiceAccount)
 }
 
-// GetServiceAccountBearerToken will attempt to get the provided service account until it
-// exists, iterate the secrets associated with it looking for one of type
-// kubernetes.io/service-account-token, and return it's token if found.
+// GetServiceAccountBearerToken determine if a ServiceAccount has a
+// bearer token sercet to use or if one should be created. It then
+// waits for the secret to have a bearer token if a secret needs to
+// be created and returns the token.
 func GetServiceAccountBearerToken(clientset kubernetes.Interface, ns string, sa string) (string, error) {
+	secretName, err := createServiceAccountTokenSecret(clientset, sa, ns)
+	if err != nil {
+		return "", err
+	}
+
 	var serviceAccount *corev1.ServiceAccount
 	var secret *corev1.Secret
-	var err error
 	err = wait.Poll(500*time.Millisecond, 30*time.Second, func() (bool, error) {
-		serviceAccount, err = clientset.CoreV1().ServiceAccounts(ns).Get(context.Background(), sa, metav1.GetOptions{})
+		secret, err = clientset.CoreV1().Secrets(ns).Get(context.Background(), secretName, metav1.GetOptions{})
```

Comment:
Agreed, any recommendation on timeout/interval? 500ms/30s was how it was originally implemented before this change.

Reply:
How about something like this:

```go
ctx, cancel := context.WithTimeout(context.Background(), 10 * time.Second)
defer cancel()
secret, err = clientset.CoreV1().Secrets(ns).Get(ctx, secretName, metav1.GetOptions{})
```

10s should be enough for even the farthest-edge cluster, and it ensures we get in at least four attempts even if the first three all time out. 

(`wait.Poll` always calls the function one more time after `timeout` is reached. It's weird, and I didn't believe it until I tested it.)

```go
// this will always take > 61 sec to run (two func calls, and one initial interval)
	_ = wait.Poll(1 * time.Second, 10 * time.Second, func() (done bool, err error) {
		time.Sleep(30 * time.Second)
		return false, nil
	})
```